### DISCUSSION TITLE
Unified Home/All/Search Page

### DISCUSSION THREAD

**Discussion by fleshgolem** [mealie-recipes/mealie#2133] [Feature Request]
Hay-Kot talked about this when doing the new search page and ever since i cant help but be somewhat annoyed that these 3 different pages that all basically do the same thing even exist
So here is my proposal on how to do this in a way that shouldn't really reduce any current functionality:

* Make "search" the new main page. Get rid of recent and all recipes
* Store last used sort order in localstorage. Maybe display a bit more prominently on mobile
* Get rid of max result setting and just use infinite scrolling as in other places
* Get rid of big search button and just do a debounced fire on all parameter changes
* Dont autofocus text-field

I think, i should be able to make most of those changes myself, but its really not something i want to get started on before putting the idea up to check if i am missing things or there are potential problems. Thoughts?

**Comment by hay-kot** [comment]
> Make "search" the new main page. Get rid of recent and all recipes

Yes

> Store last used sort order in localstorage. 

Ehhh, I'm not sure. If we move the search to be the homepage, I would think we'd _always_ want to show the most recent recipes? That keeps the current behavior. Since the Search filters are synched with the  browser URL they could always bookmark a specific search... at least that was the idea.

> Maybe display a bit more prominently on mobile

Current mobile button was a mistake, it should look like to others. 

> Get rid of max result setting and just use infinite scrolling as in other places

Yes

> Get rid of big search button and just do a debounced fire on all parameter changes

Yes, but this is tricky. Sometimes you lose state with a debounce and you're stuck in this weird awkward position.

**Comment by y8s** [comment]
Trying not to add a new discussion, but relevant to this one I think:

Allow a user to set their default landing (initial) page. Whether it's Search or Recent or All Recipes or something else. I keep finding myself come to the site and viewing 30 recipes (which is a lot for "recent" incidentally) and wondering where a recipe I remember seeing went -- THEN remembering I'm not viewing all of them.

**Reply by DrSpaldo** [reply]
@y8s totally agree. I’ve just started using Mealie and have been doing the exact same thing

**Reply by hay-kot** [reply]
This is basically how the home page works now. It's all been combined into one so I don't think this is necessary (or it's already been done)

**Reply by y8s** [reply]
@hay-kot I'm on beta 5. is it in that version or the next and if b5 then how do I make it show all?

**Reply by hay-kot** [reply]
It's currently in nightly. We're still working out a few bugs.

**Reply by DrSpaldo** [reply]
I'll try the nightly soon, I am not on the beta yet as I have been trying to work out how to install it on Unraid
